fix(docs) Moderator handbook grammar and spelling improvements (#41506)
freeCodeCamp/freeCodeCamp: docs/moderator-handbook.md

##### Ensure the CI checks are passing:

> Before merging any pull request, make sure that GitHub is reporting all checks to be passing (green checkmarks) on the pull-requests. If you see any of the checks failing, please investigate and clarify the root cause. Is the change being made breaking our tests? Will the site build correctly if the the PR is merged? These checks are critical for the stability of the platform.

Reviewer: In American English, it is `check marks` instead of `checkmarks`.
```suggestion
Before merging any pull request, make sure that GitHub is reporting all checks to be passing (green check marks) on the pull-requests. If you see any of the checks failing, please investigate and clarify the root cause. Is the change being made breaking our tests? Will the site build correctly if the the PR is merged? These checks are critical for the stability of the platform.
```

For the first spam post of a user, send them a message explaining the problem, and remove the link or post as appropriate. Leave a note on the user's profile explaining the action you have taken. If the problem persists, then quietly block the user from posting (using the silence option on the User Admin panel). Send the user a warning with the Code of Conduct. Check the box in the private message indicating that your message is a "formal warning."

You can ask questions and report incidents in the [staff forum section](https://forum.freecodecamp.com/c/staff).

### Dealing with off-topic conversations

Posts or topics that seem to be in the wrong place can be re-categorized or renamed to whatever would be appropriate.

> It may be appropriate for a moderator to fork a discussion into multiple threads in exceptional circumstances.

Reviewer: I suggest reverting this change.
```suggestion
In exceptional circumstances, it may be appropriate for a moderator to fork a discussion into multiple threads.
```

If you see anything that seems to break our [Code of Conduct](https://code-of-conduct.freecodecamp.org/), you should delete it immediately.

> Sometimes people will post things that they think are funny. They don't realize that what they said or what they shared could be interpreted as offensive. You should delete their post in these cases, but the person who posted it doesn't necessarily need to be banned. They will hopefully come to understand that what they posted was inappropriate by getting their post deleted.

Reviewer: ```suggestion
Sometimes people will post things that they think are funny. They don't realize that what they said or what they shared could be interpreted as offensive. You should delete such posts, but not necessarily ban the person. Hopefully, the user will come to understand that what they posted, was inappropriate because the post was deleted.
```

3. **Creating a Private Discussion**

   There may be situations where you need to address a concern with a camper privately. This should not be done through DMs, which can lead to situations where you claim one thing and the camper claims another. Instead, use the bot's functionality to create a private discussion:

   - Call the `!fCC private username` command, where `username` is the camper's chat user name.
   - The bot will create a new channel, and add the mentioned camper and all moderators with the `Your Friendly Moderator` role. While all moderators are added to the channel for transparency, the moderator who calls this command should be the only one to interact with the camper unless they request assistance.
   - When the conversation is complete, call the `!fCC close` command _in the private channel_ to have the bot close and delete that channel.

4. **Deleting messages**

>    Moderators can delete messages on our chat server. They should only exercise this ability in four particular situations:

Reviewer: I suggest reverting this change.
```suggestion
   Moderators can delete messages on our chat server. They should only exercise this ability in four very specific situations:
```

## How our Contributors room works

Anyone is welcome in the [Contributors room on our chat server](https://chat.freecodecamp.org/channel/contributors). It is the designated chat room for moderators and other campers who contribute to our community in any number of ways, including through study groups.

> We assume that contributors will read anything in this room that directly mentions them with an `@username`. Everything else is optional. But feel free to read anything anyone posts in there and interact.

Reviewer: ```suggestion
We assume contributors will read anything in this room that directly mentions them with an `@username`. Everything else is optional, but feel free to read anything anyone posts in there and interact.
```

Sometimes people will defend something offensive or incendiary that they said as "free speech."

> This XKCD comic perfectly summarizes most communities' thoughts on free speech. So if someone defends something, they're saying as "free speech" feel free to send it to them.

Reviewer: ```suggestion
This XKCD comic perfectly summarizes most communities' thoughts on free speech. So if someone defends something in the name of "free speech", feel free to send it to them.
```
